dapp/src/components/TransactionListener.js

```diff
@@ -57,6 +57,82 @@ class TransactionListener extends Component {
       })
       this.save(newTransactions)
     }
+
+    const nonMinedTx = this.props.transactions.filter((t) => !t.mined)
+    if (nonMinedTx.length > 0 && !this.state.wsProvider) {
+      this.startWebsocketListener()
+    } else if (nonMinedTx.length === 0 && this.state.wsProvider) {
+      this.cleanupWebSocketProvider()
```

Comment:
Should this be called in `componentWillUnmount` too?

Reply:
I don't think it is super necessary, since the listener only runs while the transaction is being mined. But it still improves the code, so adding it in. Thanks!